validator/src/main.rs

```diff
@@ -993,6 +993,7 @@ pub fn main() {
                 .takes_value(true)
                 .validator(is_parsable::<u64>)
                 .default_value(default_rocksdb_fifo_shred_storage_size)
+                .requires("rocksdb_shred_compaction")
```

Comment:
Quick question: for the `requires()`, is there a way to further specify the required argument has a certain value?
In this case, it requires `--rocksdb-shred-compaction=fifo`.

Reply:
Like this? https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.requires_if

Reply:
I think that's it!  Thank you!

Reply:
I think I might misunderstand `requires_if`.

So what we want is:  if `--rocksdb-fifo-shred-storage-size` is specified, then it requires `--rocksdb-shred-compaction=fifo`.

But from the document, it looks lik [`requires_if`](https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.requires_if) means if `--rocksdb-fifo-shred-storage-size` has the specified value, then it requires `--rocksdb-shred-compaction=fifo`.

I also checked [conflicts_with](https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.conflicts_with), but it seems it doesn't offer conditional conflict.

So I will just have `--rocksdb-fifo-shred-storage-size` requires `--rocksdb-shred-compaction`.

Reply:
Oh, sounds like you want [`required_if("rocksdb_shred_compaction", "fifo")`](https://docs.rs/clap/2.33.1/clap/struct.Arg.html#method.required_if) on `--rocksdb-fifo-shred-storage-size`, then?

Reply:
Ah, yeah so long as the arg intended to be required has a default value, it'll always be available